presto-hive/src/main/java/com/facebook/presto/hive/HiveClientModule.java

```diff
@@ -140,7 +147,14 @@ public void configure(Binder binder)
 
         configBinder(binder).bindConfig(OrcCacheConfig.class, connectorId);
 
-        binder.bind(FileOpener.class).to(HadoopFileOpener.class).in(Scopes.SINGLETON);
+        binder.bind(HiveCachingHdfsConfiguration.class).in(Scopes.SINGLETON);
```

Comment:
- This line is not necessary given you have already bound it above.
- Also, could you move `binder.bind(HdfsConfiguration.class).annotatedWith(ForMetastoreHdfsEnvironment.class).to(HiveCachingHdfsConfiguration.class).in(Scopes.SINGLETON);` to here? It is cleaner to put them side by side